### DISCUSSION TITLE
In what order should I upgrade Dagster?

### DISCUSSION THREAD

**Discussion by sryza** [dagster-io/dagster#15510] [Q&A] [answered]
Sourced from: https://dagster.slack.com/archives/C01U954MEER/p1690267219649659

We are using Dagster OSS deployment (1 dagster-daemon, 1 dagit/dagster-webserver, multiple code location) and would like to slowly upgrade from 1.3 to 1.4.

 The first issue is we've used a now removed function (reconciliation_sensor). What would be the proper way to upgrade ?
- Would it be possible to upgrade the daemon and webserver first or code location first ?
- Can code locations be in newer version than daemon / webserver without issue ?

**Comment by sryza** [comment, marked as answer]
In general, you should upgrade the host processes (daemon & webserver) before updating the user processes (code location servers). Newer host processes are guaranteed to be compatible with older user code versions, but not vice versa.

**Reply by jackwillisupside** [reply]
Is this true on major changes too? Like if I upgrade my daemon and webserver from 1.4 -> 1.5+, and my code locations are on >1.5, will that break things?
